java/code/src/com/suse/manager/reactor/hardware/HardwareMapper.java

```diff
@@ -664,19 +671,36 @@ else if (smbiosRecordsSystem.isPresent()) {
                             null, server, vCPUs, memory);
                 }
                 else {
+                    boolean isForeign = virtualInstance.getGuestSystem() != null &&
```

Comment:
Hm, aren't we interested if the *host* system of the guest is foreign? (`virtualInstance.getHostSystem()`).

Reply:
I thought you said VHM guests are foreign... would be worth double checking then indeed

Reply:
No, I probably didn't express myself clearly :/ the VHM guests are whatever the client stack runs on them (Salt or Management). Say, we have a vSphere. Then in Uyuni it'll look like this:

- vSphere registered as a Virtual Host Manager
- ESXi nodes inside vSphere appear as `FOREIGN` systems (gatherer creates them)
- virtual guests on ESXi nodes appear as `SALT` or `MANAGEMENT` (=traditional clients)

That's why it could make sense to check the _host_ system of the guest is `FOREIGN` in the quirk.

Reply:
I see... thanks for the explanation, I'll fix it.

Reply:
done